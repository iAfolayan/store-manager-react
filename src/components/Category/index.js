import React, { Component, Fragment } from 'react';
import InputForm from '../InputForms';
import { ValidateCategory } from '../../helpers/validate';
import './Category.scss';

class Category extends Component {
state = {
  name: '',
  errors:{}
}
  handleChange = event => {
    this.setState({ [event.target.name]: event.target.value});
  }

  handleSubmit = event => {
    event.preventDefault();
    const { name } = this.state;
    const errors = ValidateCategory(name);
    console.log(errors)
    if (errors) {
      this.setState({ errors });
    }
  }
  render() {
    const { name, errors } = this.state;
    return (
      <Fragment>
        <h5 className="ml-3 text-primary">Create Category</h5>
        <hr className="mr-5" />
        <div className="col-md-6 formWrapper">
          <form onSubmit={this.handleSubmit}>
              <InputForm
                error={errors.name}
                text="Name"
                label="categoryName"
                name="categoryName"
                id="categoryName"
                value={name}
                placeholder="Enter category name"
                onChange={this.handleChange}
              />
              <button className="btn btn-outline-primary float-right">Add Category</button>
          </form>
        </div>
      </Fragment>
    );
  }
}

export default Category;