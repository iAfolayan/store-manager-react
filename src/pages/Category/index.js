import React, { Fragment } from 'react';
import Footer from '../../components/Footer';
import Navigation from '../../components/Navbar';
import Category from '../../components/Category';

const CategoryHome = () => {
  return (
    <Fragment>
      <div className="form-group row">
        <div className="col-md-2">
          <Navigation />
        </div>
        <div className="col-md-10">
          <Category />
          <Footer footerClassName="footer inner" />
        </div>
      </div>
    </Fragment>
  );
}

export default CategoryHome;
