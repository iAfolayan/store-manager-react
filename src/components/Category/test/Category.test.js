import React from 'react';
import { shallow } from 'enzyme';
import Category from '..';

describe('Category component', () => {
  test('should rendered category component', () => {
    const wrapper = shallow(<Category />);
    expect(wrapper).toMatchSnapshot();
  })

  test('should simulate handleSubmit', () => {
    const wrapper = shallow(<Category />);
    wrapper.simulate().handleSubmit();
    expect(wrapper).toBeDefined();
  });
});