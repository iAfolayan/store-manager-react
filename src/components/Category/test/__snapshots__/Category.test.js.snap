// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category component should rendered category component 1`] = `
<Fragment>
  <h5
    className="ml-3 text-primary"
  >
    Create Category
  </h5>
  <hr
    className="mr-5"
  />
  <div
    className="col-md-6 formWrapper"
  >
    <form
      onSubmit={[Function]}
    >
      <InputForm
        error=""
        id="categoryName"
        label="categoryName"
        name="categoryName"
        onChange={[Function]}
        placeholder="Enter category name"
        text="Name"
        value=""
      />
      <button
        className="btn btn-outline-primary float-right"
      >
        Add Category
      </button>
    </form>
  </div>
</Fragment>
`;
